### ISSUE TITLE
sendMessage non-list contact on whatsapp-web 2.2134.9

### ISSUE THREAD

**Issue by antirek** [wppconnect-team/wppconnect#462]
`````
"TypeError: this._find is not a function
    at j._serverQuery (https://web.whatsapp.com/bootstrap_qr.abbc2ea8366e2cd9a145.js:55:334921)
    at j._query (https://web.whatsapp.com/bootstrap_qr.abbc2ea8366e2cd9a145.js:55:334450)
    at j.find (https://web.whatsapp.com/bootstrap_qr.abbc2ea8366e2cd9a145.js:55:332582)
    at Object.<anonymous> (/whatsapp_instance/node_modules/@wppconnect-team/wppconnect/dist/lib/wapi/wapi.js:1:66259)
    at Generator.next (<anonymous>)
    at qn (/whatsapp_instance/node_modules/@wppconnect-team/wppconnect/dist/lib/wapi/wapi.js:1:64935)
    at a (/whatsapp_instance/node_modules/@wppconnect-team/wppconnect/dist/lib/wapi/wapi.js:1:65139)
    at V (https://web.whatsapp.com/bootstrap_qr.abbc2ea8366e2cd9a145.js:2:373738)
    at MutationObserver.K (https://web.whatsapp.com/bootstrap_qr.abbc2ea8366e2cd9a145.js:2:373319)"
`````

![Untitled](https://user-images.githubusercontent.com/324436/131655168-b7148704-8581-4b8b-9311-b9a42a5d7da2.jpg)

**Comment by makanSukros** [comment]
window.Store.Chat._find = e => {
    const target = window.Store.Chat.get(e)
    return target ? Promise.resolve(target) : Promise.resolve({
        id: e
    })
}

solved with this

**Comment by edgardmessias** [comment]
@antirek , it is normal or multidevice version?

**Comment by antirek** [comment]
@edgardmessias it's normal version

**Comment by antirek** [comment]
it's a like https://github.com/orkestral/venom/issues/1102

**Comment by diazzaid** [comment]
> it's a like [orkestral/venom#1102](https://github.com/orkestral/venom/issues/1102)

i tried and no problem like venom..
and venom has made improvements to the update on version 3.1.4

**Comment by antirek** [comment]
maybe, you don't see problem? 
I tried send message to non-list contact and my contact don't receive message, console don't show any error. After this I run wppconnect wiith open browser and send message to non-list contact via WAPI.sendMessage() and I see error in browser console

**Comment by diazzaid** [comment]
> maybe, you don't see problem?
> I tried send message to non-list contact and my contact don't receive message, console don't show any error. After this I run wppconnect wiith open browser and send message to non-list contact via WAPI.sendMessage() and I see error in browser console

yes I tried to send a message to a number that has not been saved and that is no problem. 
make sure to use the latest version of the library

**Comment by makanSukros** [comment]
> ```
> "TypeError: this._find is not a function
>     at j._serverQuery (https://web.whatsapp.com/bootstrap_qr.abbc2ea8366e2cd9a145.js:55:334921)
>     at j._query (https://web.whatsapp.com/bootstrap_qr.abbc2ea8366e2cd9a145.js:55:334450)
>     at j.find (https://web.whatsapp.com/bootstrap_qr.abbc2ea8366e2cd9a145.js:55:332582)
>     at Object.<anonymous> (/whatsapp_instance/node_modules/@wppconnect-team/wppconnect/dist/lib/wapi/wapi.js:1:66259)
>     at Generator.next (<anonymous>)
>     at qn (/whatsapp_instance/node_modules/@wppconnect-team/wppconnect/dist/lib/wapi/wapi.js:1:64935)
>     at a (/whatsapp_instance/node_modules/@wppconnect-team/wppconnect/dist/lib/wapi/wapi.js:1:65139)
>     at V (https://web.whatsapp.com/bootstrap_qr.abbc2ea8366e2cd9a145.js:2:373738)
>     at MutationObserver.K (https://web.whatsapp.com/bootstrap_qr.abbc2ea8366e2cd9a145.js:2:373319)"
> ```
> 
> ![Untitled](https://user-images.githubusercontent.com/324436/131655168-b7148704-8581-4b8b-9311-b9a42a5d7da2.jpg)

This is because whatsapp missing find function on module

**Comment by diazzaid** [comment]
> > ```
> > "TypeError: this._find is not a function
> >     at j._serverQuery (https://web.whatsapp.com/bootstrap_qr.abbc2ea8366e2cd9a145.js:55:334921)
> >     at j._query (https://web.whatsapp.com/bootstrap_qr.abbc2ea8366e2cd9a145.js:55:334450)
> >     at j.find (https://web.whatsapp.com/bootstrap_qr.abbc2ea8366e2cd9a145.js:55:332582)
> >     at Object.<anonymous> (/whatsapp_instance/node_modules/@wppconnect-team/wppconnect/dist/lib/wapi/wapi.js:1:66259)
> >     at Generator.next (<anonymous>)
> >     at qn (/whatsapp_instance/node_modules/@wppconnect-team/wppconnect/dist/lib/wapi/wapi.js:1:64935)
> >     at a (/whatsapp_instance/node_modules/@wppconnect-team/wppconnect/dist/lib/wapi/wapi.js:1:65139)
> >     at V (https://web.whatsapp.com/bootstrap_qr.abbc2ea8366e2cd9a145.js:2:373738)
> >     at MutationObserver.K (https://web.whatsapp.com/bootstrap_qr.abbc2ea8366e2cd9a145.js:2:373319)"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ![Tanpa Judul](https://user-images.githubusercontent.com/324436/131655168-b7148704-8581-4b8b-9311-b9a42a5d7da2.jpg)
> 
> Ini karena whatsapp kehilangan fungsi find pada modul

try https://github.com/wppconnect-team/wppconnect/blob/master/examples/rest/index.js

**Comment by makanSukros** [comment]
On other library i solved with added addtional function moduleraid 
`
window.Store.Chat._find = e => {
    const target = window.Store.Chat.get(e)
    return target ? Promise.resolve(target) : Promise.resolve({
        id: e
    })
}
`

**Comment by antirek** [comment]
@diazzaid do you use whatsappVersion option in config? what value?
I set to 2.2134.9 - it's fresh whatsapp-web

**Comment by antirek** [comment]
if you use previous whatsappVersion like 2.2132.6 - all work fine

**Comment by diazzaid** [comment]
> @diazzaid apakah Anda menggunakan opsi whatsappVersion di konfigurasi? nilai apa?
> Saya set ke 2.2134.9 - ini whatsapp-web baru

i use this code
https://github.com/wppconnect-team/wppconnect/tree/master/examples/rest
![image](https://user-images.githubusercontent.com/25816482/131794421-27a18f8e-1af2-4248-b5fc-7c9d3b251f65.png)

**Comment by antirek** [comment]
oh, it's version from 21 Jul (https://github.com/wppconnect-team/wa-version/releases/tag/v1.1.3)
is it work today?

**Comment by diazzaid** [comment]
> oh, ini versi dari 21 Juli ( https://github.com/wppconnect-team/wa-version/releases/tag/v1.1.3 )
> apakah itu berfungsi hari ini?
yeah it works fine.
I don't really understand the whatsapp web version, what is clear is that I only update the library when there is a new update 
the wpp version used is the latest version 1.8.13

**Comment by edgardmessias** [comment]
> oh, it's version from 21 Jul (https://github.com/wppconnect-team/wa-version/releases/tag/v1.1.3)
> is it work today?

Yes, this project keep only "working" versions of WhatsApp.

Today I am working on it to solve this problem and another problems.

**Comment by antirek** [comment]
wa-version was updated to 2.2134.10 and this bug repeat also

**Comment by antirek** [comment]
@edgardmessias do you have the plan for test and updates for new versions of whatsapp web? 
current default version 2.2126.14. now in wa-version - 2.2132.9, 2.2136.9 and other
if i understand right that whatsapp can disable 2.2126.14 soon and wppconnect must migrate to other wa web version?

**Comment by edgardmessias** [comment]
Closed with bb13ad2b1331510c2363d0f4e29fc9c51eae49de
